plugins/anomaly-detection/AnomalyDetection.py

```diff
@@ -0,0 +1,594 @@
+            train, test = throughput_list_bxcx[0:3], throughput_list_bxcx[3:]
+            if len(test) == 1:
+                test = [test]
+            history = [x for x in train]
+            predictions = list()
+            for t in range(len(test)):
+                model = ARIMA(history, order=(1, 1, 1))
+                model_fit = model.fit()
+                output = model_fit.forecast()
+                yhat = output[0]
+                predictions.append(yhat)
+                obs = test[t]
+                history.append(obs)
+
+            predictions_final = train + predictions
+            predictions_final = inv_boxcox(predictions_final, revvar)
+            predictions_final = predictions_final.tolist()
+
+            predictions_final = [x - 1 for x in predictions_final]
+
+            return predictions_final
+        except:
+            logger.critical(
+                "Error: ARIMA encountered error with the current flow record.")
+            return None
+
+
+def calculate_arima_anomaly(dataframe):
+    """
+    The function categorizes whether a network flow is Anomalous or not based
+    on the calculated ARIMA value. A traffic is anomalous if abs(throughput
+    - arima) > Standard Deviation. True - Anomalous Traffic, False - Not
+    Anomalous
+
+    Args:
+        dataframe : The row of a dataframe containing all data related to
+        this network connection.
+    Returns:
+        A list of boolean values which signifies if a network flow record
+        of that connection is anomalous or not.
+    """
+
+    stddev = dataframe[7]
+    arima_arr = dataframe[9]
+    throughput_arr = dataframe[10]
+    anomaly_result = []
+
+    if arima_arr is None:
+        logger.error("Error: ARIMA values not calculated for this flow record")
+        result = False
+        anomaly_result.append(result)
+    elif throughput_arr is None:
+        logger.error("Error: Throughput values not in ideal format for this "
+                     "flow record")
+        result = False
+        anomaly_result.append(result)
+    else:
+        num_records = len(arima_arr)
+
+        for i in range(num_records):
+            if stddev is None:
+                logger.error("Error: Too Few Throughput Values for Standard "
+                             "Deviation to be calculated.")
+                result = False
+            elif throughput_arr[i] is None:
+                logger.error("Error: Throughput values not in ideal format "
+                             "for this flow record")
+                result = False
+            else:
+                result = True if (abs(float(throughput_arr[i]) - float(
+                    arima_arr[i])) > float(stddev)) else False
+
+            anomaly_result.append(result)
+
+    return anomaly_result
+
+
+def calculate_dbscan(diff_secs_throughput):
+    """
+    The function calculates Density-based spatial clustering of applications
```

Comment:
The comment of this function is the same as the next function `calculate_dbscan_anomaly`. I think this comment should belong to the function `calculate_dbscan_anomaly`, could you fix it?

Reply:
ack